map/extrapolation_benchmark/extrapolation_benchmark.cpp

```diff
@@ -0,0 +1,331 @@
+  if (!GetDouble(lineStream, lat))
+    return false;
+
+  return GetDouble(lineStream, lon);
+}
+
+/// \brief Fills |tracks| based on file |pathToCsv| content. File |pathToCsv| should be
+/// a text file with following lines:
+/// <Mwm name (country id)>, <Aloha id>,
+/// <Latitude of the first point>, <Longitude of the first point>, <Timestamp in seconds of the first point>,
+/// <Latitude of the second point> and so on.
+bool Parse(string const & pathToCsv, Tracks & tracks)
+{
+  tracks.clear();
+
+  std::ifstream csvStream(pathToCsv);
+  if (!csvStream.is_open())
+    return false;
+
+  while (!csvStream.eof())
+  {
+    string line;
+    getline(csvStream, line);
+    stringstream lineStream(line);
+    string dummy;
+    GetString(lineStream, dummy); // mwm id
+    GetString(lineStream, dummy); // aloha id
+
+    Track track;
+    while (!lineStream.eof())
+    {
+      double lat;
+      double lon;
+      uint64_t timestamp;
+      if (!GetGpsPoint(lineStream, timestamp, lat, lon))
+        return false;
+      track.emplace_back(static_cast<double>(timestamp), lat, lon);
+    }
+    tracks.push_back(move(track));
+  }
+  csvStream.close();
+  return true;
+}
+
+void GpsPointToGpsInfo(GpsPoint const gpsPoint, GpsInfo & gpsInfo)
+{
+  gpsInfo.m_timestamp = gpsPoint.m_timestampS;
+  gpsInfo.m_latitude = gpsPoint.m_lat;
+  gpsInfo.m_longitude = gpsPoint.m_lon;
+
+}
+} // namespace
+
+/// \brief This benchmark is written to estimate how LinearExtrapolation() extrapolates real users tracks.
+/// The idea behind the test is to measure the distance between extrapolated location and real track.
+int main(int argc, char * argv[])
+{
+  google::SetUsageMessage(
+      "Location extrapolation benchmark. Calculates mathematical expectation and dispersion for "
+      "all extrapolation deviations from tracks passed in csv with with csv_path.");
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  if (FLAGS_csv_path.empty())
+  {
+    LOG(LERROR, ("Path to csv file with user tracks should be set."));
+    return -1;
+  }
+
+  Tracks tracks;
+  if (!Parse(FLAGS_csv_path, tracks))
+  {
+    LOG(LERROR, ("An error while parsing", FLAGS_csv_path, "file. Please check if it has a correct format."));
+    return -1;
+  }
+
+  // Printing some statistics about tracks.
+  size_t trackPointNum = 0;
+  for (auto const & t : tracks)
+    trackPointNum += t.size();
+
+  double trackLengths = 0;
+  for (auto const & t : tracks)
+  {
+    double trackLenM = 0;
+    for (size_t j = 1; j < t.size(); ++j)
+    {
+      trackLenM += ms::DistanceOnEarth(ms::LatLon(t[j - 1].m_lat, t[j - 1].m_lon),
+                                       ms::LatLon(t[j].m_lat, t[j].m_lon));
+    }
+    trackLengths += trackLenM;
+  }
+
+  LOG(LINFO, ("General tracks statistics."
+              "\n  Number of tracks:", tracks.size(),
+              "\n  Number of track points:", trackPointNum,
+              "\n  Points per track in average:", trackPointNum / tracks.size(),
```

Comment:
s/in average/on average/
or
Average points per track

Reply:
Average points per track